radicle vs forges and more edits (#59)
radicle-dev/radicle-docs: docs/radicle.md

networking layer.

Since all data on the network is stored locally by peers on the network,
developers can share and collaborate on Git repositories *without* relying on
intermediaries such as hosted servers.

The radicle-link protocol distinguishes between two types of identities: personal and project.

> The first describes an actor in the system, while the second describes a (software) project
> on which one or more actors collaborate. In Radicle:

Reviewer: ```suggestion
The first describes a person running a node on the network (e.g. a peer), while the second describes a repository (e.g. a project)
on which one or more people collaborate. In Radicle: 
```

1. peers follow other peers.
2. peers track projects they are interested in.
3. peers gossip about projects. This means tracking and replicating objects from

> the peers they follow, about the projects they are interested in.

Reviewer: I'm realizing this is slightly confusing.... Isn't it more like:

1. Peers track projects they are interested in 
2. Peers track other peers by adding them as remotes to projects
3. Peers gossip about projects. This means they replicate objects from the peers they follow and make it available to the peers that follow them.

@xla

Author: Today, what you have @abbey-titcomb is more accurate. happy to adjust.

Reviewer 2: Looks good @abbey-titcomb

It's this "trusted" social graph of peers and projects that becomes the foundation for
collaboration within Radicle.

For more on Radicle Link, check out the [Understanding Radicle][wh] section.

## How is Radicle different from GitHub?

> When coming to Radicle from a centralized code collaboration platform like
> Github or Gitlab, you might notice that Radicle priotizes different concerns. Concretely:

Reviewer: This language makes it seem that Radicle is replacing GitHub as you are "coming to it". Let's rephrase the language. Also the "being in" phrase in the bullet point section reads weird.

```suggestion
When comparing Radicle to centralized code collaboration platforms like GitHub or Gitlab, it's important to note that Radicle prioritizes different concerns. Concretely:

- People should control their identity
- People should control their content
- People should control their social interactions

```

- People being in control of their identity
- People being in control of their content
- People being in control of their social interactions

Read more about the [Radicle collaboration model][fa].

> The above concerns led us to a design and implementation that differentiates itself from

Reviewer 2: ```suggestion
The concerns above led us to a design and implementation that differentiates itself from
```

> The concerns above led us to a design and implementation that differentiates itself from
> centralized forges in a few critical ways:

Reviewer: ```suggestion
Designing for these concerns differentiates Radicle from
centralized forges in a few critical ways:
```

> 1. The Radicle stack is open-source from beginning to end. There are no "closed" components.

Reviewer 2: ```suggestion
1. The Radicle stack is open-source from top to bottom. There are no "closed" components.
```

Better ring in the context of a stack.

> 1. The Radicle stack is open-source from top to bottom. There are no "closed" components.
> Every component of the Radicle stack is auditable, modifiable and extendable.

Reviewer: ```suggestion
1. The Radicle stack is completely open-source. There are no "closed" components.
Every component of the Radicle stack is auditable, modifiable, and extendable.
```

> 2. Radicle is build *entirely* on open protocols. There are no "special servers", privileged
> users or companies in control of your collaboration.

Reviewer: ```suggestion
2. Radicle is built *entirely* on open protocols. There are no "special servers", privileged
users, or companies in control of your experience.
```

> 3. Radicle is based on a peer-to-peer architecture instead of a client server model.

Reviewer: ```suggestion
3. Radicle is based on a peer-to-peer architecture instead of a client-server model.
```

4. All artifacts within Radicle are cryptographically signed and are verified by your client,

> including social artifacts like issues, PRs and comments.

Reviewer 2: This is misleading given that git objects are disseminated as-is.

Reviewer: How would you rewrite @xla ? 

Something like "All data on the Radicle network is verifiable and cryptographically signed" maybe?

Reviewer 2: My point is that we disseminate a large amount of data which is not verifiable or cryptographically signed. So I don't think we can leave this statement in, unless qualifying it, which might be more confusing than helpful as it only applies to identity data.

Author: yeah that's fair actually

> 5. Collaboration within Radicle is NOT global by default. Instead it's your "trusted"
> social graph of peers and projects that determines which content you see, interact with
> and replicate. This means that within projects, there isn't a single master branch that contributors merge into.

Reviewer: This — _This means that within projects, there isn't a single master branch that contributors merge into._ — is another point entirely

```suggestion
5. Radicle is **not** global by default. Instead, the
social graph of peers and projects you track determines what content you see, interact with,
and replicate.
```

> 6. Radicle is optimized for a local-first, dekstop experience instead of a software-as-a-service

Reviewer 2: ```suggestion
6. Radicle is optimized for a local-first experience instead of a software-as-a-service
```

Implies it is optimized for a GUI.

> 6. Radicle is optimized for a local-first experience instead of a software-as-a-service
> type of experience.

Reviewer: I don't think this comparison makes sense? Software as a service is a business model. This whole point is covered in the peer-to-peer architecture vs. the client-server model point. Let's scrap it.

```suggestion
```

> 

Reviewer: @lftherios - let's add this point instead.

```suggestion
6. Radicle's peer-to-peer design supports bazaar-style development. This means that within projects, there isn't a single master branch that contributors merge into. Instead, peers maintain their own views of projects that can be fetched and merged by other peers via patches.
```

Author: maybe "Radicle is designed for baazar-style development." and then the rest?

Reviewer: Sure! Edited.
